interfaces/apparmor/spec.go

```diff
@@ -146,6 +160,55 @@ func (spec *Specification) AddSnippet(snippet string) {
 	}
 }
 
+// AddPrioritizedSnippet adds a new apparmor snippet to all applications and hooks using the interface,
+// but identified with an UID and a priority. If no other snippet exists with that UID, the snippet is
+// added like with AddSnippet, but if there is already another snippet with that UID, the priority of
+// both will be taken into account to decide whether the new snippet replaces the old one, is appended
+// to it, or is just ignored.
+func (spec *Specification) AddPrioritizedSnippet(snippet string, uid string, priority uint) {
```

Comment:
It'd be good to avoid confusing the two arguments with some clever types and a mandatory function to construct the value perhaps? Separately uid is somewhat confusing, perhaps `key`?

```go
spec.AddSnippetKey(interfaces.SnippetKey("potato"), "...")
spec.AddSnippetKeyOverride(interfaces.SnippetKey("potato"), 1000, "...")
```

We should come up with some guidelines as to how to pick the key and the priority.

Reply:
You are right. I defined the `SnippetKey` type and used it instead of the UID. Also, added the `RegisterSnippetKey()` method to force to register a key before being able to use it as priority key. This has two advantages:

- registering the same key twice is an error, thus trying to use the same key in two different places will be detected. This is paramount to avoid a new set of privileged rules to interfere with previous rules.
- using an unregistered key is also an error, thus allowing to detect typos. This is very important too to ensure that the rules are applied as expected, and that a typo in the key doesn't break the expected results.